doc/design-component.md

```diff
@@ -0,0 +1,69 @@
+Component Loader Design
+=======================
+
+# Background
+MicroShift is designed to provide a turnkey solution that can readily serve the target workloads 
+with little intervention, probably in an airgapped environment. This requires MicroShift to
+be able to provides the most, if not all, infrastructure components during bootstrapping. 
+
+In its current form, MicroShift has an predefined asset repo and a sequence of 
+loading the manifests to the cluster. Although new assets can be added programmatically, 
+this workload clearly suffers from scalability and flexibilty. 
+
+Specifically, currently the core components are: Service-ca, OpenShift DNS, OpenShift Router, and HostPath provisioner.
+For non-interactive use cases, few of these components are required. While for Data Services oriented workloads 
+may prefer a more reliable and scale-out storage is beyond the capacity of the HostPath volume. 
+
+So it is clear that MicroShift must have a way to customize which components to load during bootstrapping and normal startup.
+
+# Existing Solutions
+
+## Operator
+Operator aims to manage the entire lifecycles of complicated components. Although Operator solves the issue
+of managing inter-dependecy, it is deemed heavier than the MicroShift use case. In many perceived occasions, the MicroShift clusters are deployed just once: they are either stay on offline mode for life (e.g. an Edge device deployed in isolated environment) or destroyed without care (e.g. a developer's test environment). There is no
+need to keep the Operator running to keep the API objects in the desired states. On the other hand, Operators' all-on pattern does not serve MicroShift's small footprint design goal.
```

Comment:
I'm not sure what "all-on pattern" means? Do you mean that Operators are continuously running their control loops, which adds load on the cluster we'd like to avoid?
Of course, users are free to install whatever however, including using Operators. I agree, though, that it's not a good default mechanism for MicroShift.

Reply:
That's what I understood. I also don't see operators as a fit for MicroShift components.